src/Persistence_matrix/include/gudhi/persistence_interval.h

```diff
@@ -88,140 +88,164 @@ struct Persistence_interval {
-
-  if constexpr (I == 0) return std::move(i.birth);
-  if constexpr (I == 1) return std::move(i.death);
-  if constexpr (I == 2) return std::move(i.dim);
+template <size_t I, typename Dimension, typename Event_value>
+constexpr auto& get(Gudhi::persistence_matrix::Persistence_interval<Dimension, Event_value>& i) noexcept {
+  return i.template get<I>();
 }
 
 /**
  * @ingroup persistence_matrix
  *
  * @brief Partial specialization of `get` for @ref Gudhi::persistence_matrix::Persistence_interval.
- *
+ * 
  * @tparam I Index of the value to return: 0 for the birth value, 1 for the death value and 2 for the dimension.
  * @tparam Dimension First template parameter of @ref Gudhi::persistence_matrix::Persistence_interval.
  * @tparam Event_value Second template parameter of @ref Gudhi::persistence_matrix::Persistence_interval.
  * @param i Interval from which the value should be returned.
  * @return Either the birth value if @p I == 0, the death value if @p I == 1 or the dimension if @p I == 2.
  */
-template <std::size_t I, typename Dimension, typename Event_value>
-constexpr const auto&& get(const Gudhi::persistence_matrix::Persistence_interval<Dimension, Event_value>&& i) noexcept {
-  static_assert(I < 3, "Value mismatch at argument 1 in template parameter list. Maximal possible value is 2.");
-
-  if constexpr (I == 0) return std::move(i.birth);
-  if constexpr (I == 1) return std::move(i.death);
-  if constexpr (I == 2) return std::move(i.dim);
+template <size_t I, typename Dimension, typename Event_value>
+constexpr const auto& get(const Gudhi::persistence_matrix::Persistence_interval<Dimension, Event_value>& i) noexcept {
+  return i.template get<I>();
 }
 
-}  // namespace persistence_matrix
-}  // namespace Gudhi
-
-namespace std {
-
 /**
  * @ingroup persistence_matrix
  *
- * @brief Partial specialization of `std::tuple_size` for @ref Gudhi::persistence_matrix::Persistence_interval.
- *
+ * @brief Partial specialization of `get` for @ref Gudhi::persistence_matrix::Persistence_interval.
+ * 
+ * @tparam I Index of the value to return: 0 for the birth value, 1 for the death value and 2 for the dimension.
  * @tparam Dimension First template parameter of @ref Gudhi::persistence_matrix::Persistence_interval.
  * @tparam Event_value Second template parameter of @ref Gudhi::persistence_matrix::Persistence_interval.
+ * @param i Interval from which the value should be returned.
+ * @return Either the birth value if @p I == 0, the death value if @p I == 1 or the dimension if @p I == 2.
  */
-template <typename Dimension, typename Event_value>
-struct tuple_size<Gudhi::persistence_matrix::Persistence_interval<Dimension, Event_value> >
-    : integral_constant<size_t, 3> {};
+template <size_t I, typename Dimension, typename Event_value>
+constexpr auto&& get(Gudhi::persistence_matrix::Persistence_interval<Dimension, Event_value>&& i) noexcept {
+  return std::move(i.template get<I>());
```

Comment:
A direct translation would have been
```suggestion
  return std::move(i).template get<I>();
```
but the result should be equivalent, so ok.